ddtrace/llmobs/_integrations/anthropic.py

```diff
@@ -0,0 +1,121 @@
+import json
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import Optional
+
+from ddtrace._trace.span import Span
+from ddtrace.internal.logger import get_logger
+from ddtrace.llmobs._constants import INPUT_MESSAGES
+from ddtrace.llmobs._constants import METADATA
+from ddtrace.llmobs._constants import MODEL_NAME
+from ddtrace.llmobs._constants import OUTPUT_MESSAGES
+from ddtrace.llmobs._constants import SPAN_KIND
+
+from .base import BaseLLMIntegration
+
+
+log = get_logger(__name__)
+
+
+API_KEY = "anthropic.request.api_key"
+MODEL = "anthropic.request.model"
+TOTAL_COST = "anthropic.tokens.total_cost"
+TYPE = "anthropic.request.type"
+
+
+class AnthropicIntegration(BaseLLMIntegration):
+    _integration_name = "anthropic"
+
+    def llmobs_set_tags(
+        self,
+        span: Span,
+        input_messages: Optional[Dict[str, Any]] = None,
+        formatted_response: Optional[Dict[str, Any]] = None,
+        err: bool = False,
+    ) -> None:
+        """Extract prompt/response tags from a completion and set them as temporary "_ml_obs.*" tags."""
+        # if not self.llmobs_enabled:
+        #     return
+        parameters = {
+            "temperature": float(span.get_tag("anthropic.request.parameters.temperature") or 1.0),
+            "max_tokens": int(span.get_tag("anthropic.request.parameters.max_tokens") or 0),
+        }
+
+        input_messages = self._extract_input_message(input_messages)
+
+        span.set_tag_str(SPAN_KIND, "llm")
+        span.set_tag_str(MODEL_NAME, span.get_tag("anthropic.request.model") or "")
+        span.set_tag_str(INPUT_MESSAGES, json.dumps(input_messages))
+        span.set_tag_str(METADATA, json.dumps(parameters))
+        if err or formatted_response is None:
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps([{"content": ""}]))
+        else:
+            output_messages = self._extract_output_message(formatted_response)
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps(output_messages))
+
+        # To-Do: record usage here as a metric of some sort?
```

Comment:
Agreed, you can follow the same pattern as `OpenAIIntegration._set_llmobs_metrics_tags()`. Note though we are setting metric names `prompt_tokens` and `completion_tokens` for Anthropic, not `input_tokens` and `output_tokens` respectively

Reply:
okay, makes sense 👍